evap/results/tools.py

```diff
@@ -48,10 +48,14 @@ def __init__(self, question, answer_counters, additional_text_result=None):
         self.additional_text_result = additional_text_result
 
         if answer_counters is not None:
-            counts = OrderedDict((value, 0) for value in self.choices.values if value != NO_ANSWER)
+            counts = OrderedDict(
+                (value, [0, name, color, value]) for (name, color, value) in self.choices.as_name_color_value_tuples()
+            )
+            counts.pop(NO_ANSWER)
             for answer_counter in answer_counters:
-                counts[answer_counter.answer] = answer_counter.count
-            self.counts = tuple(counts.values())
+                counts[answer_counter.answer][0] = answer_counter.count
+            self.zipped_choices = tuple(counts.values())
+            self.counts = tuple((count for count, _, _, _ in counts.values()))
```

Comment:
This feels overly complicated, maybe something like the following is better?
```python
            answer_indices = {
                answer: i for i, answer in enumerate(self.choices.values) if answer != NO_ANSWER
            }
            counts = [0] * len(answer_indices)
            for counter in answer_counters:
                counts[answer_indices[counter.answer]] = counter.count
            self.counts = tuple(counts)
            self.zipped_choices = self.choices.as_name_color_value_tuples()
```
We should probably refactor the whole choices business to use dataclasses instead of named tuples and increase typing with these result classes here, but we don't need to do it now

Reply:
The problem is, that `self.choices.as_name_color_value_tuples()` contains a `NO_ANSWER`-element as the last element. This needs to be removed from the result of `as_name_color_value_tuples`. Also, the counts need to be prepended